File: react_native_backup/screens/LibraryScreen.js
import React, { useState, useEffect } from "react";
import {
  StyleSheet,
  View,
  Text,
  TextInput,
  TouchableOpacity,
  ScrollView,
  SafeAreaView,
  ActivityIndicator,
} from "react-native";
import { useAudioPlayer, useAudioPlayerStatus } from "expo-audio";
import { COLORS, FONTS } from "../components/Theme";
import {
  OceanIcon,
  RainIcon,
  FireIcon,
  ForestIcon,
  SearchIcon,
  SparkleIcon,
  CompanionIcon,
} from "../components/Icons";

const CATEGORIES = ["All", "Breathing", "Grounding", "Sleep", "Focus"];

const SOUNDS = [
  { label: "Ocean", icon: (color) => <OceanIcon color={color} size={24} />, url: "https://www.soundjay.com/nature/sounds/ocean-wave-1.mp3" },
  { label: "Rain", icon: (color) => <RainIcon color={color} size={24} />, url: "https://www.soundjay.com/nature/sounds/rain-07.mp3" },
  { label: "Fire", icon: (color) => <FireIcon color={color} size={24} />, url: "https://www.soundjay.com/nature/sounds/fire-1.mp3" },
  { label: "Forest", icon: (color) => <ForestIcon color={color} size={24} />, url: "https://www.soundjay.com/nature/sounds/river-1.mp3" },
];

export default function LibraryScreen() {
  const [activeTab, setActiveTab] = useState("All");
  const [search, setSearch] = useState("");
  const player = useAudioPlayer();
  const playerStatus = useAudioPlayerStatus(player);
  const [playingSound, setPlayingSound] = useState(null); // label of currently playing sound
  const [audioLoading, setAudioLoading] = useState(false);

  const COPING_TOOLS = [
    { id: "1", icon: <OceanIcon color="#4ecdc4" size={20} />, title: "Sound Bath", subtitle: "Calming frequencies", time: "5 min", tag: "Sleep", color: "#4ecdc4" },
    { id: "2", icon: <RainIcon color="#a89af7" size={20} />, title: "Box Breathing", subtitle: "Calm your mind and reduce stress", time: "4 min", tag: "Breathing", color: "#a89af7" },
    { id: "3", icon: <ForestIcon color="#f7c59f" size={20} />, title: "Body Scan", subtitle: "Reconnect with your body", time: "10 min", tag: "Grounding", color: "#f7c59f" },
    { id: "4", icon: <SparkleIcon color="#f7a8d4" size={20} />, title: "Affirmations", subtitle: "Positive daily reminders", time: "3 min", tag: "Focus", color: "#f7a8d4" },
    { id: "5", icon: <OceanIcon color="#4ecdc4" size={20} />, title: "Grounding 5-4-3-2-1", subtitle: "Anchor yourself in the present", time: "6 min", tag: "Grounding", color: "#4ecdc4" },
    { id: "6", icon: <CompanionIcon color="#7c6ff7" size={20} />, title: "Sleep Story", subtitle: "Drift into peaceful rest", time: "20 min", tag: "Sleep", color: "#7c6ff7" },
    { id: "7", icon: <RainIcon color="#a89af7" size={20} />, title: "Mindful Breathing", subtitle: "Slow, intentional breath", time: "7 min", tag: "Breathing", color: "#a89af7" },
    { id: "8", icon: <FireIcon color="#f7e08a" size={20} />, title: "Somatic Exercises", subtitle: "Release stored tension", time: "12 min", tag: "Focus", color: "#f7e08a" },
  ];

  const handleSoundPress = async (soundItem) => {
    if (audioLoading) return;

    // Toggle off if currently playing
    if (playingSound === soundItem.label && playerStatus?.isPlaying) {
      setAudioLoading(true);
      try {
        player.pause();
        setPlayingSound(null);
      } catch (e) {
        console.log("Failed to stop audio", e);
      }
      setAudioLoading(false);
      return;
    }

    // Load new sound
    setAudioLoading(true);
    try {
      player.replace({ uri: soundItem.url });
      player.loop = true;
      player.volume = 0.8;
      player.play();
      setPlayingSound(soundItem.label);
    } catch (e) {
      console.log("Error playing audio", e);
    }
    setAudioLoading(false);
  };

  const filteredTools = COPING_TOOLS.filter((t) => {
    const matchesTab = activeTab === "All" || t.tag === activeTab;
    const matchesSearch = t.title.toLowerCase().includes(search.toLowerCase()) ||
                          t.subtitle.toLowerCase().includes(search.toLowerCase());
    return matchesTab && matchesSearch;
  });

  return (
    <SafeAreaView style={styles.container}>
      <ScrollView contentContainerStyle={styles.scrollContent} showsVerticalScrollIndicator={false}>
        {/* Header */}
        <View style={styles.header}>
          <Text style={styles.title}>Safe Space</Text>
          <Text style={styles.subtitle}>Coping Library</Text>
        </View>

        {/* Search */}
        <View style={styles.searchBar}>
          <View style={styles.searchIconBox}>
            <SearchIcon color={COLORS.muted} size={15} />
          </View>
          <TextInput
            value={search}
            onChangeText={setSearch}
            placeholder="Search tools..."
            placeholderTextColor={COLORS.muted}
            style={styles.searchInput}
          />
        </View>

        {/* Category Tabs */}
        <ScrollView
          horizontal
          showsHorizontalScrollIndicator={false}
          style={styles.tabsContainer}
          contentContainerStyle={styles.tabsContent}
        >
          {CATEGORIES.map((tab) => (
            <TouchableOpacity
              key={tab}
              onPress={() => setActiveTab(tab)}
              style={[
                styles.tabButton,
                activeTab === tab && styles.tabButtonActive,
              ]}
            >
              <Text
                style={[
                  styles.tabText,
                  activeTab === tab && styles.tabTextActive,
                ]}
              >
                {tab}
              </Text>
            </TouchableOpacity>
          ))}
        </ScrollView>

        {/* Recommended Card */}
        {search === "" && (
          <View style={styles.recommendedCard}>
            <View style={styles.recommendedBadge}>
              <Text style={styles.recommendedBadgeText}>RECOMMENDED FOR YOU</Text>
            </View>
            <View style={styles.recommendedBody}>
              <View style={styles.recommendedText}>
                <Text style={styles.recommendedTitle}>Box Breathing</Text>
                <Text style={styles.recommendedSubtitle}>Calm your mind & reduce stress</Text>
                <Text style={styles.recommendedTime}>⏱️ 4 min</Text>
              </View>
              <TouchableOpacity style={styles.playButtonCircular}>
                <Text style={styles.playButtonIcon}>▶</Text>
              </TouchableOpacity>
            </View>
          </View>
        )}

        {/* Soundboard Card */}
        <View style={[styles.card, styles.soundboardCard]}>
          <View style={styles.soundboardHeader}>
            <Text style={styles.soundboardTitle}>Sensory Reset Soundboard</Text>
            {audioLoading && <ActivityIndicator size="small" color={COLORS.tealLight} />}
          </View>
          <View style={styles.soundboardGrid}>
            {SOUNDS.map((item) => {
              const isPlaying = playingSound === item.label && playerStatus?.isPlaying;
              const activeColor = "white";
              const inactiveColor = COLORS.tealLight;
              return (
                <TouchableOpacity
                  key={item.label}
                  style={styles.soundButton}
                  activeOpacity={0.8}
                  onPress={() => handleSoundPress(item)}
                >
                  <View
                    style={[
                      styles.soundIconContainer,
                      isPlaying && styles.soundIconContainerActive,
                    ]}
                  >
                    {item.icon(isPlaying ? activeColor : inactiveColor)}
                  </View>
                  <Text
                    style={[
                      styles.soundLabel,
                      isPlaying && styles.soundLabelActive,
                    ]}
                  >
                    {item.label} {isPlaying && "🔊"}
                  </Text>
                </TouchableOpacity>
              );
            })}
          </View>
        </View>

        {/* Grid List of Tools */}
        <Text style={styles.sectionTitle}>All Coping Tools</Text>
        <View style={styles.toolsGrid}>
          {filteredTools.map((tool) => (
            <View key={tool.id} style={styles.toolCard}>
              <View style={[styles.toolIconBg, { backgroundColor: `${tool.color}12` }]}>
                {tool.icon}
              </View>
              <Text style={styles.toolTitle} numberOfLines={1}>
                {tool.title}
              </Text>
              <Text style={styles.toolSubtitle} numberOfLines={2}>
                {tool.subtitle}
              </Text>
              <View style={styles.toolFooter}>
                <Text style={styles.toolTime}>{tool.time}</Text>
                <TouchableOpacity
                  style={[styles.toolPlayBtn, { backgroundColor: `${tool.color}20` }]}
                >
                  <Text style={[styles.toolPlayIcon, { color: tool.color }]}>▶</Text>
                </TouchableOpacity>
              </View>
            </View>
          ))}
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.bg,
  },
  scrollContent: {
    paddingHorizontal: 20,
    paddingBottom: 24,
  },
  header: {
    paddingTop: 16,
    paddingBottom: 12,
  },
  title: {
    color: COLORS.white,
    fontSize: 22,
    fontWeight: "600",
    fontFamily: FONTS.serif,
  },
  subtitle: {
    color: COLORS.muted,
    fontSize: 13,
    fontFamily: FONTS.sans,
    marginTop: 2,
  },
  searchBar: {
    flexDirection: "row",
    backgroundColor: "rgba(25, 22, 54, 0.6)",
    borderRadius: 14,
    borderWidth: 1,
    borderColor: COLORS.border,
    paddingHorizontal: 12,
    paddingVertical: 10,
    alignItems: "center",
    marginBottom: 16,
  },
  searchIconBox: {
    marginRight: 8,
  },
  searchInput: {
    flex: 1,
    color: COLORS.white,
    fontSize: 13,
    fontFamily: FONTS.sans,
    paddingVertical: 0,
  },
  tabsContainer: {
    marginBottom: 16,
  },
  tabsContent: {
    gap: 8,
    paddingBottom: 4,
  },
  tabButton: {
    paddingHorizontal: 14,
    paddingVertical: 7,
    borderRadius: 20,
    backgroundColor: "rgba(25, 22, 54, 0.6)",
    borderWidth: 1,
    borderColor: COLORS.border,
  },
  tabButtonActive: {
    backgroundColor: COLORS.purple,
    borderColor: COLORS.purple,
  },
  tabText: {
    color: COLORS.muted,
    fontSize: 12,
    fontWeight: "500",
    fontFamily: FONTS.sans,
  },
  tabTextActive: {
    color: "white",
  },
  recommendedCard: {
    backgroundColor: "#112424", // Premium deep green/teal glow
    borderRadius: 18,
    borderWidth: 1,
    borderColor: "rgba(78, 205, 196, 0.22)",
    padding: 16,
    marginBottom: 20,
    shadowColor: COLORS.teal,
    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.15,
    shadowRadius: 10,
    elevation: 3,
  },
  recommendedBadge: {
    backgroundColor: "rgba(78, 205, 196, 0.1)",
    borderWidth: 1,
    borderColor: "rgba(78, 205, 196, 0.25)",
    paddingHorizontal: 8,
    paddingVertical: 4,
    borderRadius: 6,
    alignSelf: "flex-start",
    marginBottom: 10,
  },
  recommendedBadgeText: {
    color: COLORS.tealLight,
    fontSize: 9,
    fontWeight: "600",
    letterSpacing: 1,
  },
  recommendedBody: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
  },
  recommendedText: {
    flex: 1,
  },
  recommendedTitle: {
    color: COLORS.white,
    fontSize: 15,
    fontWeight: "600",
    fontFamily: FONTS.sans,
  },
  recommendedSubtitle: {
    color: COLORS.tealLight,
    fontSize: 12,
    marginTop: 2,
    fontFamily: FONTS.sans,
  },
  recommendedTime: {
    color: COLORS.muted,
    fontSize: 11,
    marginTop: 6,
    fontFamily: FONTS.sans,
  },
  playButtonCircular: {
    width: 44,
    height: 44,
    borderRadius: 22,
    backgroundColor: COLORS.teal,
    alignItems: "center",
    justifyContent: "center",
  },
  playButtonIcon: {
    color: "white",
    fontSize: 16,
    marginLeft: 2,
  },
  card: {
    backgroundColor: "rgba(19, 16, 43, 0.7)",
    borderRadius: 18,
    borderWidth: 1,
    borderColor: COLORS.border,
    padding: 16,
    marginBottom: 16,
  },
  soundboardCard: {
    backgroundColor: "rgba(13, 42, 42, 0.75)",
    borderColor: "rgba(78, 205, 196, 0.2)",
  },
  soundboardHeader: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    marginBottom: 14,
  },
  soundboardTitle: {
    color: COLORS.tealLight,
    fontSize: 11,
    fontWeight: "600",
    letterSpacing: 1,
    textTransform: "uppercase",
    fontFamily: FONTS.sans,
  },
  soundboardGrid: {
    flexDirection: "row",
    justifyContent: "space-between",
  },
  soundButton: {
    alignItems: "center",
    flex: 1,
  },
  soundIconContainer: {
    width: 52,
    height: 52,
    borderRadius: 14,
    backgroundColor: "rgba(78, 205, 196, 0.08)",
    borderWidth: 1,
    borderColor: "rgba(78, 205, 196, 0.2)",
    alignItems: "center",
    justifyContent: "center",
    marginBottom: 6,
  },
  soundIconContainerActive: {
    backgroundColor: "rgba(78, 205, 196, 0.25)",
    borderColor: COLORS.teal,
    shadowColor: COLORS.teal,
    shadowOffset: { width: 0, height: 0 },
    shadowOpacity: 0.6,
    shadowRadius: 10,
    elevation: 3,
  },
  soundLabel: {
    color: COLORS.tealLight,
    fontSize: 11,
    fontFamily: FONTS.sans,
  },
  soundLabelActive: {
    color: COLORS.white,
    fontWeight: "600",
  },
  sectionTitle: {
    color: COLORS.muted,
    fontSize: 11,
    fontWeight: "600",
    letterSpacing: 1,
    textTransform: "uppercase",
    fontFamily: FONTS.sans,
    marginVertical: 12,
  },
  toolsGrid: {
    flexDirection: "row",
    flexWrap: "wrap",
    gap: 12,
  },
  toolCard: {
    backgroundColor: "rgba(19, 16, 43, 0.7)",
    borderRadius: 16,
    borderWidth: 1,
    borderColor: COLORS.border,
    padding: 12,
    width: "47%",
    minWidth: 140,
    flexGrow: 1,
  },
  toolIconBg: {
    width: 36,
    height: 36,
    borderRadius: 10,
    alignItems: "center",
    justifyContent: "center",
    marginBottom: 10,
    borderWidth: 1,
    borderColor: "rgba(255, 255, 255, 0.03)",
  },
  toolTitle: {
    color: COLORS.white,
    fontSize: 13,
    fontWeight: "600",
    fontFamily: FONTS.sans,
    marginBottom: 2,
  },
  toolSubtitle: {
    color: COLORS.muted,
    fontSize: 11,
    fontFamily: FONTS.sans,
    lineHeight: 14,
    marginBottom: 10,
    height: 28,
  },
  toolFooter: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
  },
  toolTime: {
    color: COLORS.muted,
    fontSize: 11,
    fontFamily: FONTS.sans,
  },
  toolPlayBtn: {
    width: 26,
    height: 26,
    borderRadius: 13,
    alignItems: "center",
    justifyContent: "center",
  },
  toolPlayIcon: {
    fontSize: 10,
    marginLeft: 1,
  },
});

File: lib/screens/library_screen.dart
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import 'package:audioplayers/audioplayers.dart';
import '../theme/colors.dart';

class LibraryScreen extends StatefulWidget {
  const LibraryScreen({Key? key}) : super(key: key);

  @override
  State<LibraryScreen> createState() => _LibraryScreenState();
}

class _LibraryScreenState extends State<LibraryScreen> {
  final TextEditingController _searchController = TextEditingController();
  final AudioPlayer _audioPlayer = AudioPlayer();

  String _activeTab = 'All';
  String _searchQuery = '';
  String? _playingSoundLabel;
  bool _audioLoading = false;
  double _volume = 0.8;

  final List<String> _categories = ["All", "Breathing", "Grounding", "Sleep", "Focus"];

  final List<Map<String, dynamic>> _sounds = [
    {
      'label': 'Ocean',
      'icon': Icons.waves,
      'url': 'https://www.soundjay.com/nature/sounds/ocean-wave-1.mp3',
    },
    {
      'label': 'Rain',
      'icon': Icons.grain,
      'url': 'https://www.soundjay.com/nature/sounds/rain-07.mp3',
    },
    {
      'label': 'Fire',
      'icon': Icons.local_fire_department,
      'url': 'https://www.soundjay.com/nature/sounds/fire-1.mp3',
    },
    {
      'label': 'Forest',
      'icon': Icons.forest,
      'url': 'https://www.soundjay.com/nature/sounds/river-1.mp3',
    },
  ];

  final List<Map<String, dynamic>> _copingTools = [
    {
      'id': '1',
      'icon': Icons.waves,
      'title': 'Sound Bath',
      'subtitle': 'Calming frequencies',
      'time': '5 min',
      'tag': 'Sleep',
      'color': Color(0xFF4ECDC4)
    },
    {
      'id': '2',
      'icon': Icons.air,
      'title': 'Box Breathing',
      'subtitle': 'Calm your mind and reduce stress',
      'time': '4 min',
      'tag': 'Breathing',
      'color': Color(0xFFA89AF7)
    },
    {
      'id': '3',
      'icon': Icons.forest,
      'title': 'Body Scan',
      'subtitle': 'Reconnect with your body',
      'time': '10 min',
      'tag': 'Grounding',
      'color': Color(0xFFF7C59F)
    },
    {
      'id': '4',
      'icon': Icons.auto_awesome,
      'title': 'Affirmations',
      'subtitle': 'Positive daily reminders',
      'time': '3 min',
      'tag': 'Focus',
      'color': Color(0xFFF7A8D4)
    },
    {
      'id': '5',
      'icon': Icons.pin_drop,
      'title': 'Grounding 5-4-3-2-1',
      'subtitle': 'Anchor yourself in the present',
      'time': '6 min',
      'tag': 'Grounding',
      'color': Color(0xFF4ECDC4)
    },
    {
      'id': '6',
      'icon': Icons.nightlight_outlined,
      'title': 'Sleep Story',
      'subtitle': 'Drift into peaceful rest',
      'time': '20 min',
      'tag': 'Sleep',
      'color': Color(0xFF7C6FF7)
    },
    {
      'id': '7',
      'icon': Icons.spa,
      'title': 'Mindful Breathing',
      'subtitle': 'Slow, intentional breath',
      'time': '7 min',
      'tag': 'Breathing',
      'color': Color(0xFFA89AF7)
    },
    {
      'id': '8',
      'icon': Icons.self_improvement,
      'title': 'Somatic Exercises',
      'subtitle': 'Release stored tension',
      'time': '12 min',
      'tag': 'Focus',
      'color': Color(0xFFF7E08A)
    },
  ];

  @override
  void initState() {
    super.initState();
    _audioPlayer.setReleaseMode(ReleaseMode.loop);
    _audioPlayer.setVolume(_volume);
    _audioPlayer.onPlayerStateChanged.listen((state) {
      if (state == PlayerState.playing) {
        setState(() {
          _audioLoading = false;
        });
      }
    });
  }

  @override
  void dispose() {
    _audioPlayer.dispose();
    _searchController.dispose();
    super.dispose();
  }

  Future<void> _handleSoundPress(Map<String, dynamic> soundItem) async {
    final label = soundItem['label'] as String;
    final url = soundItem['url'] as String;

    if (_audioLoading) return;

    if (_playingSoundLabel == label) {
      setState(() => _audioLoading = true);
      try {
        await _audioPlayer.stop();
        setState(() {
          _playingSoundLabel = null;
          _audioLoading = false;
        });
      } catch (e) {
        print("Failed to stop audio: $e");
        setState(() => _audioLoading = false);
      }
      return;
    }

    setState(() {
      _audioLoading = true;
      _playingSoundLabel = label;
    });

    try {
      await _audioPlayer.stop();
      await _audioPlayer.setVolume(_volume);
      await _audioPlayer.setReleaseMode(ReleaseMode.loop);
      await _audioPlayer.play(UrlSource(url));
    } catch (e) {
      print("Error playing audio: $e");
      setState(() {
        _playingSoundLabel = null;
        _audioLoading = false;
      });
    }
  }

  Future<void> _changeVolume(double val) async {
    setState(() {
      _volume = val;
    });
    await _audioPlayer.setVolume(val);
  }

  @override
  Widget build(BuildContext context) {
    // Filter tools
    final filteredTools = _copingTools.where((t) {
      final matchesTab = _activeTab == 'All' || t['tag'] == _activeTab;
      final matchesSearch = t['title'].toString().toLowerCase().contains(_searchQuery.toLowerCase()) ||
          t['subtitle'].toString().toLowerCase().contains(_searchQuery.toLowerCase());
      return matchesTab && matchesSearch;
    }).toList();

    return Scaffold(
      backgroundColor: Colors.transparent,
      body: SingleChildScrollView(
        physics: const BouncingScrollPhysics(),
        padding: const EdgeInsets.symmetric(horizontal: 20.0, vertical: 12.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            // Header
            Text(
              'Safe Space',
              style: GoogleFonts.playfairDisplay(
                fontSize: 22,
                fontWeight: FontWeight.w600,
                color: AppColors.textPrimary,
              ),
            ),
            const SizedBox(height: 2),
            Text(
              'Coping Library',
              style: GoogleFonts.dmSans(
                fontSize: 13,
                color: AppColors.muted,
              ),
            ),
            const SizedBox(height: 16),

            // Search Bar
            _buildSearchBar(),
            const SizedBox(height: 16),

            // Category Tab Scroll
            _buildCategoryTabs(),
            const SizedBox(height: 16),

            // Recommended Card
            if (_searchQuery.isEmpty) _buildRecommendedCard(),
            if (_searchQuery.isEmpty) const SizedBox(height: 20),

            // Soundboard Card
            _buildSoundboard(),
            const SizedBox(height: 20),

            // All Coping Tools Header
            Text(
              'ALL COPING TOOLS',
              style: GoogleFonts.dmSans(
                fontSize: 11,
                fontWeight: FontWeight.bold,
                color: AppColors.muted,
                letterSpacing: 1.0,
              ),
            ),
            const SizedBox(height: 12),

            // Grid List of Tools
            _buildToolsGrid(filteredTools),
            const SizedBox(height: 24),
          ],
        ),
      ),
    );
  }

  Widget _buildSearchBar() {
    return Container(
      decoration: BoxDecoration(
        color: AppColors.bgCard.withOpacity(0.6),
        borderRadius: BorderRadius.circular(14),
        border: Border.all(color: AppColors.border),
      ),
      padding: const EdgeInsets.symmetric(horizontal: 14),
      child: Row(
        children: [
          const Icon(Icons.search, color: AppColors.muted, size: 16),
          const SizedBox(width: 8),
          Expanded(
            child: TextField(
              controller: _searchController,
              onChanged: (val) => setState(() => _searchQuery = val),
              style: GoogleFonts.dmSans(color: AppColors.textPrimary, fontSize: 13.5),
              decoration: InputDecoration(
                hintText: 'Search tools...',
                hintStyle: GoogleFonts.dmSans(color: AppColors.muted),
                border: InputBorder.none,
                isDense: true,
              ),
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildCategoryTabs() {
    return SizedBox(
      height: 38,
      child: ListView.builder(
        scrollDirection: Axis.horizontal,
        physics: const BouncingScrollPhysics(),
        itemCount: _categories.length,
        itemBuilder: (context, index) {
          final tab = _categories[index];
          final isActive = _activeTab == tab;

          return GestureDetector(
            onTap: () => setState(() => _activeTab = tab),
            child: Container(
              margin: const EdgeInsets.only(right: 8),
              padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 8),
              decoration: BoxDecoration(
                color: isActive ? AppColors.purple : AppColors.bgCard.withOpacity(0.6),
                borderRadius: BorderRadius.circular(20),
                border: Border.all(
                  color: isActive ? AppColors.purple : AppColors.border,
                ),
              ),
              child: Text(
                tab,
                style: GoogleFonts.dmSans(
                  fontSize: 12,
                  fontWeight: FontWeight.w500,
                  color: isActive ? AppColors.textPrimary : AppColors.muted,
                ),
              ),
            ),
          );
        },
      ),
    );
  }

  Widget _buildRecommendedCard() {
    return Container(
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: const Color(0xFF112424), // Premium deep green/teal glow
        borderRadius: BorderRadius.circular(18),
        border: Border.all(color: AppColors.teal.withOpacity(0.22)),
        boxShadow: [
          BoxShadow(
            color: AppColors.teal.withOpacity(0.15),
            blurRadius: 10,
            offset: const Offset(0, 4),
          )
        ],
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Container(
            padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
            decoration: BoxDecoration(
              color: AppColors.teal.withOpacity(0.1),
              borderRadius: BorderRadius.circular(6),
              border: Border.all(color: AppColors.teal.withOpacity(0.25)),
            ),
            child: Text(
              'RECOMMENDED FOR YOU',
              style: GoogleFonts.dmSans(
                fontSize: 9,
                fontWeight: FontWeight.bold,
                color: AppColors.tealLight,
                letterSpacing: 1.0,
              ),
            ),
          ),
          const SizedBox(height: 10),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      'Box Breathing',
                      style: GoogleFonts.dmSans(
                        fontSize: 15,
                        fontWeight: FontWeight.bold,
                        color: AppColors.textPrimary,
                      ),
                    ),
                    const SizedBox(height: 2),
                    Text(
                      'Calm your mind & reduce stress',
                      style: GoogleFonts.dmSans(
                        fontSize: 12,
                        color: AppColors.tealLight,
                      ),
                    ),
                    const SizedBox(height: 6),
                    Text(
                      '⏱ 4 min',
                      style: GoogleFonts.dmSans(
                        fontSize: 11,
                        color: AppColors.muted,
                      ),
                    ),
                  ],
                ),
              ),
              Container(
                width: 44,
                height: 44,
                decoration: const BoxDecoration(
                  shape: BoxShape.circle,
                  color: AppColors.teal,
                ),
                child: const Icon(Icons.play_arrow, color: AppColors.textPrimary, size: 20),
              )
            ],
          )
        ],
      ),
    );
  }

  Widget _buildSoundboard() {
    return Container(
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: const Color(0xCC0D2A2A), // deep green/teal background
        borderRadius: BorderRadius.circular(18),
        border: Border.all(color: AppColors.teal.withOpacity(0.2)),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Text(
                'SENSORY RESET SOUNDBOARD',
                style: GoogleFonts.dmSans(
                  fontSize: 11,
                  fontWeight: FontWeight.bold,
                  color: AppColors.tealLight,
                  letterSpacing: 1.0,
                ),
              ),
              if (_audioLoading)
                const SizedBox(
                  width: 12,
                  height: 12,
                  child: CircularProgressIndicator(strokeWidth: 2, color: AppColors.tealLight),
                )
            ],
          ),
          const SizedBox(height: 14),

          // Sound Grid
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: _sounds.map((sound) {
              final label = sound['label'] as String;
              final icon = sound['icon'] as IconData;
              final isPlaying = _playingSoundLabel == label;

              return Expanded(
                child: GestureDetector(
                  onTap: () => _handleSoundPress(sound),
                  child: Column(
                    children: [
                      AnimatedContainer(
                        duration: const Duration(milliseconds: 250),
                        width: 52,
                        height: 52,
                        decoration: BoxDecoration(
                          color: isPlaying ? AppColors.teal.withOpacity(0.25) : AppColors.teal.withOpacity(0.08),
                          borderRadius: BorderRadius.circular(14),
                          border: Border.all(
                            color: isPlaying ? AppColors.teal : AppColors.teal.withOpacity(0.2),
                          ),
                          boxShadow: isPlaying
                              ? [
                                  BoxShadow(
                                    color: AppColors.teal.withOpacity(0.4),
                                    blurRadius: 10,
                                  )
                                ]
                              : null,
                        ),
                        child: Icon(
                          icon,
                          color: isPlaying ? AppColors.textPrimary : AppColors.tealLight,
                          size: 22,
                        ),
                      ),
                      const SizedBox(height: 6),
                      Text(
                        '$label ${isPlaying ? "🔊" : ""}',
                        style: GoogleFonts.dmSans(
                          fontSize: 11,
                          fontWeight: isPlaying ? FontWeight.bold : FontWeight.normal,
                          color: isPlaying ? AppColors.textPrimary : AppColors.tealLight,
                        ),
                        maxLines: 1,
                        overflow: TextOverflow.ellipsis,
                      ),
                    ],
                  ),
                ),
              );
            }).toList(),
          ),

          // Live Volume Slider (shown when sound is playing)
          if (_playingSoundLabel != null) ...[
            const SizedBox(height: 16),
            const Divider(color: AppColors.textPrimary10),
            Row(
              children: [
                const Icon(Icons.volume_down, color: AppColors.tealLight, size: 16),
                Expanded(
                  child: SliderTheme(
                    data: SliderTheme.of(context).copyWith(
                      activeTrackColor: AppColors.teal,
                      inactiveTrackColor: AppColors.textPrimary.withOpacity(0.06),
                      trackHeight: 4.0,
                      thumbColor: AppColors.teal,
                      thumbShape: const RoundSliderThumbShape(enabledThumbRadius: 5.0),
                      overlayColor: AppColors.teal.withOpacity(0.12),
                    ),
                    child: Slider(
                      value: _volume,
                      min: 0.0,
                      max: 1.0,
                      onChanged: _changeVolume,
                    ),
                  ),
                ),
                const Icon(Icons.volume_up, color: AppColors.tealLight, size: 16),
              ],
            ),
          ]
        ],
      ),
    );
  }

  Widget _buildToolsGrid(List<Map<String, dynamic>> tools) {
    if (tools.isEmpty) {
      return Center(
        child: Padding(
          padding: const EdgeInsets.symmetric(vertical: 40.0),
          child: Text(
            'No tools match your query.',
            style: GoogleFonts.dmSans(color: AppColors.muted, fontSize: 13),
          ),
        ),
      );
    }

    return GridView.builder(
      shrinkWrap: true,
      physics: const NeverScrollableScrollPhysics(),
      gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
        crossAxisCount: 2,
        childAspectRatio: 0.9,
        crossAxisSpacing: 12,
        mainAxisSpacing: 12,
      ),
      itemCount: tools.length,
      itemBuilder: (context, index) {
        final tool = tools[index];
        final Color color = tool['color'] as Color;

        return Container(
          padding: const EdgeInsets.all(12),
          decoration: BoxDecoration(
            color: AppColors.bgCard.withOpacity(0.7),
            borderRadius: BorderRadius.circular(16),
            border: Border.all(color: AppColors.border),
          ),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Container(
                width: 36,
                height: 36,
                decoration: BoxDecoration(
                  color: color.withOpacity(0.08),
                  borderRadius: BorderRadius.circular(10),
                  border: Border.all(color: AppColors.textPrimary.withOpacity(0.03)),
                ),
                child: Icon(tool['icon'] as IconData, color: color, size: 20),
              ),
              const SizedBox(height: 10),
              Text(
                tool['title'] as String,
                style: GoogleFonts.dmSans(
                  fontSize: 13,
                  fontWeight: FontWeight.bold,
                  color: AppColors.textPrimary,
                ),
                maxLines: 1,
                overflow: TextOverflow.ellipsis,
              ),
              const SizedBox(height: 2),
              Expanded(
                child: Text(
                  tool['subtitle'] as String,
                  style: GoogleFonts.dmSans(
                    fontSize: 11,
                    color: AppColors.muted,
                    height: 1.3,
                  ),
                  maxLines: 2,
                  overflow: TextOverflow.ellipsis,
                ),
              ),
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  Text(
                    tool['time'] as String,
                    style: GoogleFonts.dmSans(fontSize: 11, color: AppColors.muted),
                  ),
                  Container(
                    width: 26,
                    height: 26,
                    decoration: BoxDecoration(
                      shape: BoxShape.circle,
                      color: color.withOpacity(0.12),
                    ),
                    child: Icon(Icons.play_arrow, color: color, size: 12),
                  )
                ],
              )
            ],
          ),
        );
      },
    );
  }
}
